### PR TITLE
make culling work with nb prefix, culling timout per notebook

### PR DESCRIPTION
**Pull request by nikp1172**

### CODE REVIEW

File: notebook-controller/controllers/culling_controller.go [hunk 1]
```diff
@@ -186,33 +188,52 @@ func notebookIsIdle(meta metav1.ObjectMeta, log logr.Logger) bool {
...
 	client := &http.Client{
 		Timeout: time.Second * 10,
 	}
-
+	var nb_prefix = "/"
+	nb_prefix_annotation, ok := meta.GetAnnotations()[TFY_NB_PREFIX_ANNOTATION]
+	if ok {
+		nb_prefix = nb_prefix_annotation
+	} else {
+		log.Info(fmt.Sprintf("NB_PREFIX annotation: %s not found in notebook, using NB_PREFIX=/", TFY_NB_PREFIX_ANNOTATION))
```
Comment (srihari-tf) [inline notebook-controller/controllers/culling_controller.go:224]: should we use the default one instead /notebook/<ns>/<nb>

**Review by srihari-tf — APPROVED** [review]

File: notebook-controller/controllers/culling_controller.go (hunk 1 above)
Comment (nikp1172) [reply notebook-controller/controllers/culling_controller.go:224]: default NB_PREFIX is /notebook/ws/ns I think

Comment (nikp1172) [reply notebook-controller/controllers/culling_controller.go:224]: But whatever is after the domain name is NB_PREFIX. So if we want it to be available there, it means that it has to be available at '/'

Comment (srihari-tf) [reply notebook-controller/controllers/culling_controller.go:230]: no the thing is this annotation might be missing if we do not set it during deployment - so if the notebook is deployed in some other way, we should support the default NB_PREFIX

**Pull request merged**